lib/src/compiler/linters.rs

```diff
@@ -114,6 +115,145 @@ impl LinterInternal for RuleName {
 
 type Predicate<'a> = dyn Fn(&Meta) -> bool + 'a;
 
+/// A linter that ensures tags meet specified requirements in either an allowed
+/// list of tags or in a regex.
+///
+/// ```
+/// # use yara_x::Compiler;
+/// use yara_x::linters;
+/// let mut compiler = Compiler::new();
+/// let warnings = compiler
+///     .add_linter(linters::tags_allowed(vec!["foo".to_string(), "bar".to_string()]))
+///     // This produces a warning because the rule tags are not from the
+///     // allowed list
+///     .add_source(r#"rule foo : test { strings: $foo = "foo" condition: $foo }"#)
+///     .unwrap()
+///     .warnings();
+///
+/// assert_eq!(
+///     warnings[0].to_string(),
+///     r#"warning[unknown_tag]: Tag not in allowed list
+///  --> line:1:12
+///   |
+/// 1 | rule foo : test { strings: $foo = "foo" condition: $foo }
+///   |            ---- tag `test` not in allowed list
+///   |
+///   = note: Allowed tags: foo, bar"#);
+pub struct Tags {
+    allow_list: Vec<String>,
+    regex: String,
```

Comment:
I would make this an `Option<String>` and initialize it with `None`. That's more efficient that allocating an empty string during initialization, and also suggests that this string is optional, which is the case.